### PR TITLE
Fetch the db connection in json format

### PR DESCRIPTION
**Pull request by FritzOnFire**
We now fetch the db connection details from config service in json format. This allows us to have a more dynamic approach to handling the db connection details.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -51,84 +52,89 @@ var validDBNameRegex = regexp.MustCompile(`^[_\-a-zA-Z0-9]+$`)
...
+		s += fmt.Sprintf(" port=%v", c.Port)
 	}
-	if c.password != "" {
-		s += fmt.Sprintf(" password=%v", c.password)
+	if c.Password != "" {
+		s += fmt.Sprintf(" password=%v", c.Password)
 	}
-	if c.sslmode != "" {
-		s += fmt.Sprintf(" sslmode=%v", c.sslmode)
+	if c.SSLMode != "" && c.SSLMode != "allow" { // allow is not an accepted value for sslmode
```
Comment (bosvos) [inline main.go:74]: Could you elaborate why "allow" would not be acceptable?

File: main.go [hunk 2]
```diff
@@ -51,84 +52,89 @@ var validDBNameRegex = regexp.MustCompile(`^[_\-a-zA-Z0-9]+$`)
...
 
 // postgres:hostname:port:dbname:username:password
 // port and password may be blank, in which case they are omitted from the connection string
 // Returns driver, con, error
-func parseDBConStr(dbStr string) (dbCon, error) {
+func parseDBConStr(dbStr string) (*dbCon, error) {
 	parts := strings.Split(dbStr, ":")
-	if len(parts) != 7 {
-		return dbCon{}, fmt.Errorf("Invalid db connection string. Expected 6 colon-separated parts, but only got %v parts", len(parts))
+	if len(parts) > 5 {
```
Comment (bosvos) [inline main.go]: Are you sure this is correct? Surely it should be <6 ?

File: main.go [hunk 3]
```diff
@@ -345,33 +346,42 @@ func runMigrations(log *log.Logger, dbStr string, sqlFiles []string) error {
...
+		return nil, err
 	}
 	if resp.StatusCode != 200 {
-		return "", fmt.Errorf("%v %v", resp.StatusCode, string(body))
+		return nil, fmt.Errorf("%v %v", resp.StatusCode, string(body))
+	}
+
+	var con dbCon
+	if err := json.Unmarshal(body, &con); err != nil {
+		return nil, fmt.Errorf("while marshalling the response from config service to get the db connection details: %v", err)
```
Comment (nkosimarz) [inline main.go]: While

File: main.go (hunk 1 above)
Comment (FritzOnFire) [reply main.go:74]: This is what the driver told me. There does not seem to be an alternative.
```
pq: unsupported sslmode "allow"; only "require" (default), "verify-full", "verify-ca", and "disable" supported
```

**Review by nkosimarz — APPROVED** [review]
LGTM, pending other comment on this pr regarding len(parts)

**Pull request merged**